packages/frontend/core/src/components/pure/cmdk/highlight.tsx

```diff
@@ -24,33 +17,85 @@ type HighlightLabelProps = {
   highlight: string;
 };
 
+function useHighlight(text: string, query: string) {
```

Comment:
i think it is better to have a `highlightTextFragments` function before wrapping it as a hook, and then write a unit test for `highlightTextFragments` alone.